libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,647 @@
+            event.selectedItems = this.selectedItems;
+            this.stateChanges.next(event);
+            this.selectedItemChange.emit(event);
+
+        });
+
+
+    }
+
+    /** @hidden */
+    /**Keyboard manager on list items, set values when passed via array */
+    ngAfterViewInit(): void {
+
+        this.keyManager = new FocusKeyManager<BaseListItem>(this.ListItems).withWrap();
+        this.ListItems.forEach((item) => {
+            item.navigated = this.navigated;
+            item.navigationIndicator = this.navigationIndicator;
+            item.contentDensity = this.contentDensity;
+            item.selectionMode = this.selectionMode;
+            item.listType = this.listType;
+            item.hasByLine = this.hasByLine;
+            item.noSeperator = this.noSeperator;
+            this.stateChanges.next(item);
+        });
+
+    }
+
+
+    /** @hidden */
+    /**Setting values from list to list items
+     * example:
+     * Does list item has navigation,
+     * should show arrows,
+     * will it be compact mode,
+     * should be in which selection mode
+     * set values when passed via datasource
+     */
+    ngAfterContentInit(): void {
+        this._itemsSubscription = this.ListItems.changes.subscribe((items) => {
+            items.forEach((item) => {
+                item.navigated = this.navigated;
+                item.navigationIndicator = this.navigationIndicator;
+                item.contentDensity = this.contentDensity;
+                item.selectionMode = this.selectionMode;
+                item.listType = this.listType;
+                item.hasByLine = this.hasByLine;
+                item.noSeperator = this.noSeperator;
+                this.stateChanges.next(item);
+            });
+
+        });
+    }
+
+    ngOnDestroy(): void {
+        super.ngOnDestroy();
+
+        if (isDataSource(this.dataSource)) {
+            this.dataSource.close();
+        }
+        if (this._dsSubscription) {
+            this._dsSubscription.unsubscribe();
+        }
+        if (this.selectionModel) {
+            this.selectionModel.changed.unsubscribe();
+        }
+        if (this._itemsSubscription) {
+            this._itemsSubscription.unsubscribe();
+        }
+    }
+
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+
+    set value(value: any) {
+        super.setValue(value);
+    }
+
+
+    /** @hidden */
+    /**handline keyboard operations
+    *  on list and list items
+    */
+    handleKeyDown(event: any): boolean {
+        event.stopImmediatePropagation();
+        if (this.keyManager) {
+            if (event.keyCode === DOWN_ARROW || event.keyCode === UP_ARROW) {
+                return false;
+            } else if (event.keyCode === ENTER || event.keyCode === SPACE) {
+                this.updateNavigation(event);
+                return false;
+            }
+        }
+    }
+
+    private initializeDS(ds: FdpListDataSource<any>): void {
```

Comment:
Let's add `_` prefixes to private properties

Reply:
done